platform/android/src/text/local_glyph_rasterizer.cpp

```diff
@@ -0,0 +1,124 @@
+#include <mbgl/text/local_glyph_rasterizer.hpp>
+#include <mbgl/util/i18n.hpp>
+#include <mbgl/util/platform.hpp>
+
+#include <jni/jni.hpp>
+
+#include "../attach_env.hpp"
+#include "../bitmap.hpp"
+
+#include "local_glyph_rasterizer_impl.hpp"
+
+/*
+    Android implementation of LocalGlyphRasterizer:
+     Draws CJK glyphs using locally available fonts.
+
+    Follows pattern of GL JS implementation in that:
+     - Only CJK glyphs are drawn locally (because we can guess their metrics effectively)
+        * Render size/metrics determined experimentally using Noto Sans
+     - Configuration is done at map creation time by setting a "font family"
+        * JS uses a CSS font-family, this uses android.graphics.Typeface
+          https://developer.android.com/reference/android/graphics/Typeface.html
+     - We use heuristics to extract a font-weight based on the incoming font stack
+        * JS tries to extract multiple weights, this implementation only looks for
+          "bold"
+
+     mbgl::LocalGlyphRasterizer is the portable interface
+     mbgl::LocalGlyphRasterizer::Impl stores platform-specific configuration data
+     mbgl::android::LocalGlyphRasterizer is the JNI wrapper
+     com.mapbox.mapboxsdk.text.LocalGlyphRasterizer is the Java implementation that
+      actually does the drawing
+ */
+
+namespace mbgl {
+namespace android {
+
+PremultipliedImage LocalGlyphRasterizer::drawGlyphBitmap(const std::string& fontFamily, const bool bold, const GlyphID glyphID) {
+    UniqueEnv env { AttachEnv() };
+
+    using Signature = jni::Object<Bitmap>(jni::String, jni::jboolean, jni::jchar);
+    auto method = javaClass.GetStaticMethod<Signature>(*env, "drawGlyphBitmap");
+
+    jni::String jniFontFamily = jni::Make<jni::String>(*env, fontFamily);
+
+    auto result = javaClass.Call(*env,
```

Comment:
Have you tried this with many calls? I wonder if we should delete the local reference here to avoid local reference table overflows.

Reply:
@ChrisLoer the size limit may be as low as 512. See #10528.

Reply:
Pushed a change based on @asheemmamoowala 's guidance.